libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/SalesforceLogger.java

```diff
@@ -0,0 +1,293 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.analytics.logger;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A simple logger that allows components to log statements of different log levels. This class
+ * also provides the ability to break down logs at the component level and set different log
+ * levels for different components. The available options for log output are console and file.
+ *
+ * @author bhariharan
+ */
+public class SalesforceLogger {
+
+    private static final String TAG = "SalesforceLogger";
+    private static final String LOG_LINE_FORMAT = "LEVEL: %s, TAG: %s, MESSAGE: %s";
+    private static final String LOG_LINE_FORMAT_WITH_EXCEPTION = "LEVEL: %s, TAG: %s, MESSAGE: %s, EXCEPTION: %s";
+    private static final ExecutorService THREAD_POOL = Executors.newFixedThreadPool(3);
+    private static Map<String, SalesforceLogger> LOGGERS;
+
+    /**
+     * An enumeration of log levels.
+     */
+    public enum Level {
+        OFF,
+        ERROR,
+        WARN,
+        INFO,
+        DEBUG,
+        VERBOSE
+    }
+
+    private FileLogger fileLogger;
+    private Context context;
+    private Level logLevel;
+
+    /**
+     * Returns a logger instance associated with a named component.
+     *
+     * @param componentName Component name.
+     * @param context Context.
+     * @return Logger instance.
+     */
+    public synchronized static SalesforceLogger getLogger(String componentName, Context context) {
+        if (LOGGERS == null) {
+            LOGGERS = new HashMap<>();
+        }
+        if (!LOGGERS.containsKey(componentName)) {
+            final SalesforceLogger logger = new SalesforceLogger(componentName, context);
+            LOGGERS.put(componentName, logger);
+        }
+        return LOGGERS.get(componentName);
+    }
+
+    /**
+     * Returns the set of components that have loggers associated with them.
+     *
+     * @return Set of components being logged.
+     */
+    public synchronized static Set<String> getComponents() {
+        if (LOGGERS == null || LOGGERS.size() == 0) {
+            return null;
+        }
+        Set<String> components = LOGGERS.keySet();
+        if (components.size() == 0) {
+            components = null;
+        }
+        return components;
+    }
+
+    /**
+     * Wipes all components currently being logged. Used only by tests.
+     */
+    public synchronized static void flushComponents() {
+        LOGGERS = null;
+    }
+
+    private SalesforceLogger(String componentName, Context context) {
+        this.context = context;
+        if (isDebugMode()) {
```

Comment:
If the `APK` is a debug binary, the default level is `DEBUG`. If it's a release binary, the default level is `ERROR`.